Summaries/CSTA/testing/configuration.txt
Configurations
{'K': 200,
 'audio_feature_path': '/home/yejin/data/dataset/MR.HiSum/feature_h5/audio.h5',
 'aux_loss': True,
 'aux_loss_coef': 0,
 'batch_size': 32,
 'best_f1score_save_dir': 'Summaries/CSTA/testing/best_f1score_model',
 'best_ktau_save_dir': 'Summaries/CSTA/testing/best_ktau_model',
 'best_map15_save_dir': 'Summaries/CSTA/testing/best_mAP15_model',
 'best_map50_save_dir': 'Summaries/CSTA/testing/best_mAP50_model',
 'best_srho_save_dir': 'Summaries/CSTA/testing/best_srho_model',
 'ckpt_path': None,
 'clamp': False,
 'contrastive_align_loss': False,
 'contrastive_align_loss_coef': 0.02,
 'contrastive_hdim': 64,
 'dec_layers': 2,
 'dec_loss_coef': 1,
 'denoiser': 'DiT',
 'device': device(type='cuda', index=0),
 'dim_feedforward': 1024,
 'dropout': 0.1,
 'dropout_ratio': 0.5,
 'early_stop_by': 'None',
 'ema': False,
 'ema_decay': 0.99,
 'enc_layers': 2,
 'eos_coef': 0.1,
 'epochs': 100,
 'eps': 0.01,
 'gamma': 0.99,
 'giou_loss_coef': 1,
 'gt_path': '/home/yejin/data/projects/yejin/VideoSum/Triplesumm/data/mrsum_with_features_gtsummary_modified.h5',
 'hidden_dim': 256,
 'input_dim': 768,
 'input_dropout': 0.5,
 'l2_reg': 1e-07,
 'label_loss_coef': 4,
 'lr': 5e-05,
 'lw_saliency': 4.0,
 'max_v_l': -1,
 'mix_type': 'v',
 'model': 'CSTA',
 'n_input_proj': 2,
 'nheads': 8,
 'num_scores': 300,
 'p_uncond': 0.0,
 'position_embedding': 'sine',
 'pre_norm': False,
 'saliency_margin': 0.2,
 'save_dir_root': 'Summaries/CSTA/testing',
 'save_results': True,
 'scores_embed': 'learned',
 'set_cost_class': 4,
 'set_cost_giou': 1,
 'set_cost_span': 10,
 'sigmoid_temp': 1.0,
 'span_loss_coef': 10,
 'span_loss_type': 'l1',
 'split_file_path': '/home/yejin/data/projects/yejin/VideoSum/Triplesumm/data/custom_split.json',
 'tag': 'testing',
 'temperature': 0.07,
 'tensorboard_path': '/home/yejin/data/projects/yejin/VideoSum/Triplesumm/tensorboard',
 'text_feature_path': '/home/yejin/data/dataset/MR.HiSum/feature_h5/transcripts.h5',
 'timestamp_data_path': '/home/yejin/data/dataset/MR.HiSum/feature_h5/timestamp.h5',
 'train': True,
 'vis_feature_path': '/home/yejin/data/dataset/MR.HiSum/feature_h5/video.h5',
 'w': 0.1}





Summaries/A2Summ/testing/configuration.txt
Configurations
{'audio_feature_path': '/home/yejin/data/dataset/MR.HiSum/feature_h5/audio.h5',
 'batch_size': 32,
 'best_f1score_save_dir': 'Summaries/A2Summ/testing/best_f1score_model',
 'best_ktau_save_dir': 'Summaries/A2Summ/testing/best_ktau_model',
 'best_map15_save_dir': 'Summaries/A2Summ/testing/best_mAP15_model',
 'best_map50_save_dir': 'Summaries/A2Summ/testing/best_mAP50_model',
 'best_srho_save_dir': 'Summaries/A2Summ/testing/best_srho_model',
 'ckpt_path': None,
 'device': 'cuda',
 'dropout_attn': 0.1,
 'dropout_fc': 0.5,
 'dropout_text': 0.1,
 'dropout_video': 0.1,
 'early_stop_by': 'None',
 'epochs': 1,
 'gt_path': '/home/yejin/data/projects/yejin/VideoSum/Triplesumm/data/mrsum_with_features_gtsummary_modified.h5',
 'input_dim_text': 768,
 'input_dim_vis': 768,
 'lambda_contrastive_inter': 0.0,
 'lambda_contrastive_intra': 0.0,
 'lr': 5e-05,
 'mix_type': 'vt',
 'model': 'A2Summ',
 'num_hidden': 256,
 'num_layers': 2,
 'num_workers': 4,
 'ratio': 4,
 'save_dir_root': 'Summaries/A2Summ/testing',
 'save_results': True,
 'seed': 12345,
 'split_file_path': '/home/yejin/data/projects/yejin/VideoSum/Triplesumm/data/custom_split.json',
 'tag': 'testing',
 'tensorboard_path': '/home/yejin/data/projects/yejin/VideoSum/Triplesumm/tensorboard',
 'text_feature_path': '/home/yejin/data/dataset/MR.HiSum/feature_h5/transcripts.h5',
 'timestamp_data_path': '/home/yejin/data/dataset/MR.HiSum/feature_h5/timestamp.h5',
 'train': True,
 'vis_feature_path': '/home/yejin/data/dataset/MR.HiSum/feature_h5/video.h5',
 'weight_decay': 1e-07}





networks/csta/config.py
import json 
import argparse
import numpy as np


def build_config():
    from configs import Config, str2bool
    parser = argparse.ArgumentParser("This script is used for PGL_SUM/CSTA summarization task.")
    
    # Training & evaluation
    parser.add_argument('--model', type = str, default = 'MLP', help = 'the name of the model')
    parser.add_argument('--epochs', type = int, default = 200, help = 'the number of training epochs')
    parser.add_argument('--lr', type = float, default = 5e-5, help = 'the learning rate')
    parser.add_argument('--l2_reg', type = float, default = 1e-4, help = 'l2 regularizer')
    parser.add_argument('--dropout_ratio', type = float, default = 0.5, help = 'the dropout ratio')
    parser.add_argument('--gamma', type=float, default=0.99)

    parser.add_argument('--batch_size', type = int, default = 256, help = 'the batch size')
    parser.add_argument('--tag', type = str, default = 'dev', help = 'A tag for experiments')
    parser.add_argument('--ckpt_path', type = str, default = None, help = 'checkpoint path for inference or weight initialization')
    parser.add_argument('--train', type=str2bool, default='true', help='when use Train')
    parser.add_argument('--save_results', type=bool, default=False, help='when save results')
    parser.add_argument('--early_stop_by', type=str, default=None)
    parser.add_argument('--tensorboard_path', type = str, default = '/home/yejin/data/projects/yejin/VideoSum/Triplesumm/tensorboard')

    # Data Config
    parser.add_argument('--vis_feature_path', type=str, help="path to visual feature h5 file")
    parser.add_argument('--text_feature_path', type=str, help="path to text feature h5 file")
    parser.add_argument('--audio_feature_path', type=str, help="path to audio feature h5 file")
    parser.add_argument('--timestamp_data_path', type=str, help="path to timestamp feature h5 file")
    parser.add_argument('--gt_path', type=str, help="path to gt h5 file")
    parser.add_argument('--split_file_path', type=str, help="path to split file")
    parser.add_argument("--max_v_l", type=int, default=-1)
    parser.add_argument('--mix_type', type=str, default='v')
    parser.add_argument('--input_dim', type=int, default=768)
    

    # Model Config
    parser.add_argument('--position_embedding', default='sine', type=str, choices=('sine', 'learned'), help="Type of positional embedding to use on top of the image features")
    parser.add_argument('--enc_layers', default=2, type=int, help="Number of encoding layers in the transformer")
    parser.add_argument('--dec_layers', default=2, type=int, help="Number of decoding layers in the transformer")
    parser.add_argument('--hidden_dim', default=256, type=int, help="Size of the embeddings (dimension of the transformer)")
    parser.add_argument('--dropout', default=0.1, type=float, help="Dropout applied in the transformer")
    parser.add_argument('--nheads', default=8, type=int, help="Number of attention heads inside the transformer's attentions")
    parser.add_argument('--dim_feedforward', default=1024, type=int, help="Intermediate size of the feedforward layers in the transformer blocks")
    parser.add_argument('--pre_norm', action='store_true')
    parser.add_argument('--num_scores', default=300, type=int, help="Number of query slots")
    parser.add_argument('--input_dropout', default=0.5, type=float, help="Dropout applied in input")
    parser.add_argument("--n_input_proj", type=int, default=2, help="#layers to encoder input")
    parser.add_argument("--temperature", type=float, default=0.07, help="temperature nce contrastive_align_loss")
    parser.add_argument("--K", type=int, default=200, help="Quantization size for scores")
    parser.add_argument("--denoiser", type=str, default='DiT', choices=['DiT', 'Transformer_dec', 'latentmlp'], help="Denoiser model")
    parser.add_argument("--p_uncond", type=float, default=0.0, help="Probability of sampling from unconditional")
    parser.add_argument("--w", type=float, default=0.1, help="weight for unconditional sampling")
    parser.add_argument("--ema", type=str2bool, default=False, help="use EMA for denoiser")
    parser.add_argument("--ema_decay", type=float, default=0.99, help="decay for EMA")
    parser.add_argument("--sigmoid_temp", type=float, default=1.0, help="temperature for sigmoid")
    parser.add_argument("--eps", type=float, default=1e-2, help="eps for logit")
    parser.add_argument("--scores_embed", type=str, default='learned', choices=['learned', 'sinusoidal'], help="score embedding")
    parser.add_argument("--clamp", type=str2bool, default=False, help="clamp scores")

    # Loss
    parser.add_argument('--no_aux_loss', dest='aux_loss', action='store_false', help="Disables auxiliary decoding losses (loss at each layer)")
    parser.add_argument("--contrastive_align_loss", action="store_true", help="Disable contrastive_align_loss between matched query spans and the text.")
    parser.add_argument("--contrastive_hdim", type=int, default=64, help="dim for contrastive embeddings")
    parser.add_argument("--span_loss_type", default="l1", type=str, choices=['l1', 'ce'], 
                        help="l1: (center-x, width) regression. ce: (st_idx, ed_idx) classification.")
    parser.add_argument("--lw_saliency", type=float, default=4.,
                            help="weight for saliency loss, set to 0 will ignore")
    parser.add_argument("--saliency_margin", type=float, default=0.2)
        
        
    # * Matcher
    parser.add_argument('--set_cost_span', default=10, type=float, help="L1 span coefficient in the matching cost")
    parser.add_argument('--set_cost_giou', default=1, type=float, help="giou span coefficient in the matching cost")
    parser.add_argument('--set_cost_class', default=4, type=float, help="Class coefficient in the matching cost")

    # Loss coefficientsw
    parser.add_argument('--span_loss_coef', default=10, type=float)
    parser.add_argument('--giou_loss_coef', default=1, type=float)
    parser.add_argument('--label_loss_coef', default=4, type=float)
    parser.add_argument('--eos_coef', default=0.1, type=float,
                        help="Relative classification weight of the no-object class")
    parser.add_argument("--contrastive_align_loss_coef", default=0.02, type=float)
    parser.add_argument("--aux_loss_coef", default=0, type=float)
    parser.add_argument("--dec_loss_coef", default=1, type=float)

    args = parser.parse_args()

    kwargs = vars(args)
    config = Config(**kwargs)

    return config